internal/pool/pool.go

```diff
@@ -249,22 +251,23 @@ func (p *ConnPool) getTurn() {
 }
 
 func (p *ConnPool) waitTurn(ctx context.Context) error {
-	var done <-chan struct{}
-	if ctx != nil {
-		done = ctx.Done()
+	if ctx == nil {
+		// The background context's Done is nil by default.
+		ctx = context.Background()
 	}
```

Comment:
Please revert

Reply:
Why should this be reverted?  This version of the code is cleaner.

The existing code actually had a latent nil pointer bug due to how it was structured.